DALK.PL_ANALYZER/Models/Matches/GroupStage.cs

```diff
@@ -7,18 +7,11 @@ namespace DALK.PL_ANALYZER.Models.Matches
 {
     public class GroupStage : IStage
     {
-        private byte round;
-        private byte maxRound;
-
-        public GroupStage(byte round, byte maxRound)
-        {
-            this.round = round;
-            this.maxRound = maxRound;
-        }
-
+        public byte Round { get; set; }
```

Comment:
From my experience if better to have immutable objects. Private set to the rescue